src/other/utils.jl

```diff
@@ -58,3 +58,42 @@ function gennames(n::Integer)
     return res
 end
 
+"""
+    flatten(df::AbstractDataFrame, veccol::Union{Integer, Symbol})
+
+When column `veccol` of `df` has elements non-zero length, for example a `Vector` 
+of `Vector`s. Returns a DataFrame where each element of `veccol` is flattened. 
+Elements of row `i` of `df` other than `veccol` will be duplicated according to 
+the length of `df[i, veccol]`
+
+**Arguments**
+
+* `df`: An `AbstractDataFrame`
+* `veccol`: A `Symbol` or `Integer` where `df[:, veccol]` is a column whose 
+elements support iteration.
+"""
+
+function Base.Iterators.flatten(df::AbstractDataFrame, veccol::Union{Integer, Symbol})
```

Comment:
I'm not sure that it's correct for this to be a method of `Iterators.flatten`. To me, this is a fundamentally different operation (un-nesting an iterable versus un-nesting a single iterable field). Since `flatten` is in `Base.Iterators`, you can even use the `flatten` function-name without collision.

Reply:
I think you are correct. JuliaDB does not extend Base.Iterators.flatten. I have deleted the extension.

Reply:
But then we should introduce it to DataAPI.jl and mutually extend in DataFrames.jl and JuliaDB.jl (CC @nalimilan).